src/View/AjaxView.php

```diff
@@ -95,39 +103,70 @@ public function render($view = null, $layout = null) {
 		}
 
 		if ($view !== false && !isset($this->viewVars['_redirect']) && $this->_getViewFileName($view)) {
-			$response['content'] = parent::render($view, $layout);
+			$dataToSerialize['content'] = parent::render($view, $layout);
 		}
+
+		$this->viewVars = Hash::merge($this->viewVars, $dataToSerialize);
+
 		if (isset($this->viewVars['_serialize'])) {
-			$response = $this->_serialize($response, $this->viewVars['_serialize']);
+			$dataToSerialize = $this->_dataToSerialize($this->viewVars['_serialize'], $dataToSerialize);
 		}
-		$result = json_encode($response);
+
+		return $this->_serialize($dataToSerialize);
+	}
+
+	/**
+	 * Serialize(json_encode) accumulated data from both our custom render method
+	 *   and viewVars set by the user.
+	 *
+	 * @param array $dataToSerialize Array of data that is to be serialzed.
+	 * @return array The serialized data.
+	 */
+	protected function _serialize($dataToSerialize = []) {
+		$result = json_encode($dataToSerialize);
 		if (json_last_error() !== JSON_ERROR_NONE) {
 			return json_encode(['error' => json_last_error_msg()]);
 		}
 		return $result;
 	}
 
 	/**
-	 * Serializes view vars.
+	 * Returns data to be serialized based on the value of viewVars.
 	 *
-	 * @param array $response Response data array.
-	 * @param array $serialize The viewVars that need to be serialized.
-	 * @return array The serialized data.
+	 * @param array|string|bool $serialize The name(s) of the view variable(s) that
+	 *   need(s) to be serialized. If true all available view variables will be used.
+	 * @param array $additionalData Data items that were defined internally in our own
+	 *   render method.
+	 * @return mixed The data to serialize.
 	 */
-	protected function _serialize($response, $serialize) {
+	protected function _dataToSerialize($serialize = true, $additionalData = []) {
```

Comment:
You dont need default value for $serialize, as you always pass a value in the calling part.